pkg/web_app/lib/src/gtm_js.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+@JS()
+library gtm_js;
+
+import 'dart:html' as dart_html;
+
+import 'package:js/js.dart';
+import 'package:js/js_util.dart';
+
+@JS('dataLayer.push')
+external void _push(data);
+
+void _pushMap(Map<String, dynamic> data) {
+  _push(jsify(data));
+}
+
+/// Records a GTM [action] with the default [event] name and [category].
+void gtmCustomEventClick({
+  String event = 'custom-event',
```

Comment:
This should not be something that can be changed, we should just hardcode it.
It's also hardcoded in the trigger in google tag manager.